subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/annotations/NestedBeanAnnotationHandler.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.properties.annotations;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.codehaus.groovy.runtime.ConvertedClosure;
+import org.gradle.api.internal.tasks.DefaultTaskInputPropertySpec;
+import org.gradle.api.internal.tasks.PropertySpecFactory;
+import org.gradle.api.internal.tasks.TaskValidationContext;
+import org.gradle.api.internal.tasks.ValidatingValue;
+import org.gradle.api.internal.tasks.ValidationAction;
+import org.gradle.api.internal.tasks.properties.NestedBeanContext;
+import org.gradle.api.internal.tasks.properties.NestedBeanResolver;
+import org.gradle.api.internal.tasks.properties.PropertyNode;
+import org.gradle.api.internal.tasks.properties.PropertyValue;
+import org.gradle.api.internal.tasks.properties.PropertyVisitor;
+import org.gradle.api.tasks.Nested;
+import org.gradle.internal.UncheckedException;
+
+import javax.annotation.Nullable;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+
+import static org.gradle.api.internal.tasks.TaskValidationContext.Severity.ERROR;
+
+public class NestedBeanAnnotationHandler implements PropertyAnnotationHandler {
+
+    private NestedBeanResolver<PropertyNode> nestedBeanResolver = new NestedBeanResolver<PropertyNode>();
```

Comment:
This should have been `final` to begin with. It could have at least been `static`, too, to match the other use of this type. But actually, the type should be turned into a `static` utility type instead.

Reply:
It is a static method now.